<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows():void.ac3563291d194d478a70de5ac9422ac8_node6</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">6</a></h1></center>PROC: <a href="./../org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8.html">void AssumptionTest.assumeNoExceptionThrows()</a> LINE: <a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>
<br>PREDS:
<a href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node5.html"><span class='visited with_tooltip'>N_5<span class='tooltip'>node5 preds:4 succs:6 exn:14 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node11.html"><span class='visited with_tooltip'>N_11<span class='tooltip'>node11 preds:10 6 succs:2 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node14.html"><span class='visited with_tooltip'>N_14<span class='tooltip'>node14 preds: succs:12 13 exn:3 </span></span></a><br>
<LISTING class='color_green'>n$4=_fun_void Assert.fail(String)("Should have thrown exception":java.lang.Object*) [line 147];
</LISTING>
<hr width="100%">
<a name="session20" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session20">node6#session20</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [compute pre] liveness 6
<LISTING class='color_black'>JOIN

LEFT:   normal:{ }, exn:{ }
RIGHT:  normal:{ exception, return }, exn:{ }
RESULT: normal:{ exception, return }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session21" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session21">node6#session21</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] liveness 6
<LISTING class='color_black'>PRE STATE:
normal:{ exception, return }, exn:{ }

INSTR=  n$4=_fun_void Assert.fail(String)(&quot;Should have thrown exception&quot;:java.lang.Object*) [line 147]; 

STATE:
normal:{ exception, return }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session33" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session33">node6#session33</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session34" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session34">node6#session34</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ exception }, exn:{ },normal:{ n$2, n$3 }, exn:{ })

INSTR=  n$4=_fun_void Assert.fail(String)(&quot;Should have thrown exception&quot;:java.lang.Object*) [line 147]; 

STATE:
(normal:{ exception }, exn:{ },normal:{ n$4 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session60" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session60">node6#session60</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [compute pre] devirtualizer analysis 6
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session61" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session61">node6#session61</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] devirtualizer analysis 6
<LISTING class='color_black'>PRE STATE:
{ n$0 -&gt; java.lang.NullPointerException,
  n$2 -&gt; java.lang.NullPointerException,
  exception -&gt; java.lang.NullPointerException }

INSTR=  n$4=_fun_void Assert.fail(String)(&quot;Should have thrown exception&quot;:java.lang.Object*) [line 147]; 

STATE UNCHANGED

INSTR=  EXIT_SCOPE(n$4); [line 147]; 

STATE UNCHANGED

INSTR=  APPLY_ABSTRACTION; [line 147]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session104" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session104">node6#session104</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session105" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session105">node6#session105</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session106" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session106">node6#session106</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session6" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session6">node6#session6</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [compute pre] starvation
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session7" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session7">node6#session7</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] starvation
<LISTING class='color_black'>PRE STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  n$4=_fun_void Assert.fail(String)(&quot;Should have thrown exception&quot;:java.lang.Object*) [line 147]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  EXIT_SCOPE(n$4); [line 147]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

INSTR=  APPLY_ABSTRACTION; [line 147]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session17" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session17">node6#session17</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [compute pre] racerd
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session18" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session18">node6#session18</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) [exec] racerd
<LISTING class='color_black'>PRE STATE:
Threads: NoThread, Locks: 0 
Accesses { } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Owned,
             n$3 -&gt; Unowned,
             exception -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 


INSTR=  n$4 := void Assert.fail(String)({ &quot;Should have thrown exception&quot; }) [line 147]; 

STATE:
Threads: NoThread, Locks: 0 
Accesses { } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Owned,
             n$3 -&gt; Unowned,
             n$4 -&gt; Owned,
             exception -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 


INSTR=  EXIT_SCOPE(n$4); [line 147]; 

STATE UNCHANGED

INSTR=  APPLY_ABSTRACTION; [line 147]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session26" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session26">node6#session26</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void AssumptionTest.assumeNoExceptionThrows() Node: 6, Procedure: void AssumptionTest.assumeNoExceptionThrows(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
  <span class='color_red'>exception = _t$5<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar20 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar1 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_red'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
  [footprint
     <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$4=_fun_void Assert.fail(String)("Should have thrown exception":java.lang.Object*) [line 147];
  EXIT_SCOPE(n$4); [line 147];
  APPLY_ABSTRACTION; [line 147];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$4=_fun_void Assert.fail(String)("Should have thrown exception":java.lang.Object*) [line 147]</span>
    in vpath_find exp:n$33
    vpath_find: found exception :java.lang.Throwable*
    Found 2 specs for function org.junit.Assert.fail(java.lang.String):void
    START EXECUTING SPECS FOR org.junit.Assert.fail(java.lang.String):void from state
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    EXECUTING SPEC 1/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
      [footprint
         this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>message|callee = null<span class='color_orange'>formal</span>:java.lang.String*( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; new:87 &gt; <span class='color_orange'>{vpath: }</span>(_t$52)</span> ;
      <span class='color_red'>return|callee = EXN _t$52<span class='color_orange'>update:87</span>:void</span> ;
      <span class='color_red'>_t$52|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(n$33); RET &lt; NullPointerException.<init>() &gt; (n$39)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    |-
    <span class='color_blue'>message|callee = null<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    
      do_imply "Should have thrown exception" null
      WARNING: footprint failed to find MISSING because: constants not equal
    EXECUTING SPEC 2/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
      [footprint
         this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>message|callee != null ;
      message|callee = _t$53<span class='color_orange'>formal</span>:java.lang.String*( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'>message|callee != null; <span class='color_red'>MEMne &lt; new:89 &gt; <span class='color_orange'>{vpath: }</span>(_t$54)</span> ;
      <span class='color_red'>return|callee = EXN _t$54<span class='color_orange'>update:89</span>:void</span> ;
      <span class='color_red'>_t$54|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(n$33); RET &lt; NullPointerException.<init>() &gt; (n$39)
    pi2:
      _t$53 != null
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    |-
    <span class='color_blue'>message|callee = _t$53<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    
      do_imply "Should have thrown exception" _t$53
        Final Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
        exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
        [footprint
           this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
        [SUB: 
          _t$53 = "Should have thrown exception"]
        |-
        <span class='color_blue'></span>
    imply_pi: adding missing atom _t$53 != null
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException
      [footprint
         this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
      [SUB: 
        _t$53 = "Should have thrown exception"
      _t$53 != null]
      |-
      <span class='color_blue'>"Should have thrown exception" != null ;
      message|callee = "Should have thrown exception"<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    returning TRUE
    in vpath_find exp:_t$54
    vpath_find: cannot find _t$54
      New footprint:
      "Should have thrown exception" != null
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      n$33|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; &amp;exception|->n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; &amp;CatchVar20|->n$34<span class='color_orange'>initial</span>:void; &amp;$irvar0|->n$35<span class='color_orange'>initial</span>:void; &amp;return|->n$36<span class='color_orange'>initial</span>:void; &amp;$irvar1|->n$37<span class='color_orange'>initial</span>:void; &amp;this|->@f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); &amp;old_this|->@f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); &amp;e|->n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>"Should have thrown exception" != null</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$54)</span> ;
      <span class='color_red'>return|callee = EXN _t$54<span class='color_orange'>update:147</span>:void</span> ;
      <span class='color_red'>_t$54|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>null != "Should have thrown exception"</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (n$39)</span> ;
    <span class='color_blue'>exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = n$34<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = n$35<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = n$37<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    [footprint
       <span class='color_red'>null != "Should have thrown exception"</span> *
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>null != "Should have thrown exception"</span>; <span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:144</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_red'>null != "Should have thrown exception"</span> *
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$4); [line 147]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>null != "Should have thrown exception"</span>; <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:147</span>:void</span>; <span class='color_blue'>$irvar1 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>null != "Should have thrown exception"</span> *
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 147]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0) ;
    exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = _t$2<span class='color_orange'>initial</span>:void; $irvar0 = _t$3<span class='color_orange'>initial</span>:void; return = EXN _t$0<span class='color_orange'>update:147</span>:void; $irvar1 = _t$4<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )  ]</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:147</span>:void</span>; <span class='color_blue'>$irvar1 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:144</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session37" href="./../nodes/org.junit.tests.experimental.AssumptionTest.assumeNoExceptionThrows%28%29%3Avoid.ac3563291d194d478a70de5ac9422ac8_node6.html#session37">node6#session37</a>(<a href="./../../AssumptionTest.java.fd33b35ae585dc55015f0b3438e646ba.html#LINE147">147</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssumptionTest.assumeNoExceptionThrows() Node: 6, Procedure: void AssumptionTest.assumeNoExceptionThrows(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
  <span class='color_red'>exception = _t$5<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar20 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar1 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$4=_fun_void Assert.fail(String)("Should have thrown exception":java.lang.Object*) [line 147];
  EXIT_SCOPE(n$4); [line 147];
  APPLY_ABSTRACTION; [line 147];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$4=_fun_void Assert.fail(String)("Should have thrown exception":java.lang.Object*) [line 147]</span>
    in vpath_find exp:n$33
    vpath_find: found exception :java.lang.Throwable*
    Found 2 specs for function org.junit.Assert.fail(java.lang.String):void
    START EXECUTING SPECS FOR org.junit.Assert.fail(java.lang.String):void from state
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    EXECUTING SPEC 1/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>message|callee = null<span class='color_orange'>formal</span>:java.lang.String*( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; new:87 &gt; <span class='color_orange'>{vpath: }</span>(_t$52)</span> ;
      <span class='color_red'>return|callee = EXN _t$52<span class='color_orange'>update:87</span>:void</span> ;
      <span class='color_red'>_t$52|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(n$33); RET &lt; NullPointerException.<init>() &gt; (n$39)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    |-
    <span class='color_blue'>message|callee = null<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    
      do_imply "Should have thrown exception" null
      WARNING: footprint failed to find MISSING because: constants not equal
    EXECUTING SPEC 2/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>message|callee != null ;
      message|callee = _t$53<span class='color_orange'>formal</span>:java.lang.String*( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'>message|callee != null; <span class='color_red'>MEMne &lt; new:89 &gt; <span class='color_orange'>{vpath: }</span>(_t$54)</span> ;
      <span class='color_red'>return|callee = EXN _t$54<span class='color_orange'>update:89</span>:void</span> ;
      <span class='color_red'>_t$54|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(n$33); RET &lt; NullPointerException.<init>() &gt; (n$39)
    pi2:
      _t$53 != null
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
    exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    |-
    <span class='color_blue'>message|callee = _t$53<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    
      do_imply "Should have thrown exception" _t$53
        Final Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
        exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
        [SUB: 
          _t$53 = "Should have thrown exception"]
        |-
        <span class='color_blue'></span>
    imply_pi: adding missing atom _t$53 != null
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); RET &lt; NullPointerException.<init>() &gt; (n$39) ;
      exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = n$34<span class='color_orange'>initial</span>:void; message|callee = "Should have thrown exception"<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar0 = n$35<span class='color_orange'>initial</span>:void; return = n$36<span class='color_orange'>initial</span>:void; $irvar1 = n$37<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
      [SUB: 
        _t$53 = "Should have thrown exception"
      _t$53 != null]
      |-
      <span class='color_blue'>"Should have thrown exception" != null ;
      message|callee = "Should have thrown exception"<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>
    returning TRUE
    in vpath_find exp:_t$54
    vpath_find: cannot find _t$54
      New footprint:
      "Should have thrown exception" != null
      Frame fld:
      
      Missing fld:
      
      Instantiated frame:
      n$33|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; &amp;exception|->n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*; &amp;CatchVar20|->n$34<span class='color_orange'>initial</span>:void; &amp;$irvar0|->n$35<span class='color_orange'>initial</span>:void; &amp;return|->n$36<span class='color_orange'>initial</span>:void; &amp;$irvar1|->n$37<span class='color_orange'>initial</span>:void; &amp;this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); &amp;old_this|->val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); &amp;e|->n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>"Should have thrown exception" != null</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$54)</span> ;
      <span class='color_red'>return|callee = EXN _t$54<span class='color_orange'>update:147</span>:void</span> ;
      <span class='color_red'>_t$54|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>null != "Should have thrown exception"</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (n$39)</span> ;
    <span class='color_blue'>exception = n$33<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = n$34<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = n$35<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = n$37<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = n$38<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Missing pure facts for the function call:
    pi: "Should have thrown exception" != null
    Prover.check_inconsistency_base: inconsistency reason [Some 2]
    Found minimum cover
    pi: "Should have thrown exception" != null
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>null != "Should have thrown exception"</span>; <span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:144</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$4); [line 147]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>null != "Should have thrown exception"</span>; <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:147</span>:void</span>; <span class='color_blue'>$irvar1 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>APPLY_ABSTRACTION; [line 147]</span>
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Throwable*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    <span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0) ;
    exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*; CatchVar20 = _t$2<span class='color_orange'>initial</span>:void; $irvar0 = _t$3<span class='color_orange'>initial</span>:void; return = EXN _t$0<span class='color_orange'>update:147</span>:void; $irvar1 = _t$4<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub ); e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:144</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:147</span>:void</span>; <span class='color_blue'>$irvar1 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = _t$5<span class='color_orange'>initial</span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:144 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; void Assert.fail(String):147 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:144</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar20 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:147</span></span>:void</span>; <span class='color_blue'>$irvar1 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.experimental.AssumptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:org.junit.AssumptionViolatedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    
    
  </LISTING>